Vagrantfile

```diff
@@ -0,0 +1,52 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+  config.vm.box = "hashicorp/bionic64"
+  
+  # Increase instance size to improve performance of installing python version below
+  config.vm.provider "virtualbox" do |v|
+    v.memory = 4096
+    v.cpus = 2
+  end
+  
+  # Forward host port to app Flask port on VM
+  config.vm.network "forwarded_port", guest: 5000, host: 5000
+  
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL
+    sudo apt-get update
+    
+    # Install pyenv prerequisites
+    sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+    
+    # Install pyenv
+    rm -r ~/.pyenv
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    echo 'echo "Path:" $PATH' >> ~/.profile
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
+    echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
+    echo 'eval "$(pyenv init -)"' >> ~/.profile
+    
+    # Install Python 3.8.5
+    echo 'python --version' >> ~/.profile
+    echo 'pyenv install 3.8.5' >> ~/.profile
+    echo 'pyenv global 3.8.5' >> ~/.profile
+    echo 'python --version' >> ~/.profile
+    source ~/.profile
+  SHELL
+  
+  config.trigger.after :up do |trigger|
+    trigger.name = "Launching App"
+    trigger.info = "Running the TODO app setup script"
+    trigger.run_remote = {privileged: false, inline: "
+      # Install dependencies and launch
+      cd /vagrant
+      pip install -r requirements.txt
+      flask run --host=0.0.0.0
```

Comment:
This works, but there are two changes that would be nice.

- Poetry is a better dependency manager than pip, so it would be good if you switch your project to use that. (We are going to update the exercise brief to actually ask learners to use poetry)
- Currently this makes the `vagrant up` command hang. The biggest issue with that is vagrant manager gets confused and you can't e.g. `vagrant halt`. It would be nicer to run it in a background process. This can be achieved with `nohup yourcommand > logfile.txt 2>&1 &`. Breaking it down: `nohup ... &` runs a command as a separate process, `... > logfile.txt` redirects "stdout" to a text file so you can read what would have been printed to the terminal. `2>&1` redirects stderr to stdout so errors also get logged.